app/src/main/java/org/liberty/android/fantastischmemo/ui/CardMCEditor.java

```diff
@@ -87,7 +91,8 @@ public void onCreate(Bundle savedInstanceState){
         dbOpenHelper = AnyMemoDBOpenHelperManager.getHelper(getApplicationContext(), dbPath);
         multipleChoiceCardDao = dbOpenHelper.getMultipleChoiceDao();
         multipleChoiceCardDao.setHelper(dbOpenHelper);
-        updateViews();
+        setCard(currentMCCard);
+        updateCard(cardToUpdate);
```

Comment:
Why do you have to set the card before updating it? Can't you just update he currentMCCard?

Reply:
setCard function takes the values that are written in the fields of the ui and updates the card attributes. Ill change it to a more meaningful name. updateCard actually just displays the inital values of the card in the fields of the ui so when you want to edit a card the info is already displayed.